--- openssl-1.1.1f/crypto/rand/rand_unix.c
+++ openssl-1.1.1f/crypto/rand/rand_unix.c
@@ -364,7 +364,7 @@ static ssize_t syscall_random(void *buf, size_t buflen)
      * - Linux since 3.17 with glibc 2.25
      * - FreeBSD since 12.0 (1200061)
      */
-#  if defined(__GNUC__) && __GNUC__>=2 && defined(__ELF__) && !defined(__hpux)
+#  if defined(__GNUC__) && __GNUC__>=2 && defined(__ELF__) && !defined(__hpux) && !defined(__use_urandom)
     extern int getentropy(void *buffer, size_t length) __attribute__((weak));

     if (getentropy != NULL)
--
2.26.2

